docs/custom-filters.adoc

```diff
@@ -442,33 +443,37 @@ public class FetchFilter implements FetchRequestFilter {
                                                                RequestHeaderData header,
                                                                FetchRequestData request,
                                                                FilterContext context) {
-        MetadataRequestData metadataRequest = new MetadataRequestData(); //<1>
+        var metadataRequestHeader = new RequestHeaderData().setRequestApiVersion(METADATA_VERSION_SUPPORTING_TOPIC_IDS); // <1>
+        var metadataRequest = new MetadataRequestData(); // <2>
         var topic = new MetadataRequestData.MetadataRequestTopic();
         topic.setTopicId(Uuid.randomUuid());
         metadataRequest.topics().add(topic);
-        short version = METADATA_VERSION_SUPPORTING_TOPIC_IDS;
-        CompletionStage<MetadataResponseData> stage = context.sendRequest(version, request); //<2>
-        return stage.thenApply(metadataResponse -> mutateFetchRequest(metadataResponse, request)) // <3>
-                    .thenCompose(mutatedFetchRequest -> context.forwardRequest(header, mutatedFetchRequest)); // <4>
+        var stage = context.sendRequest(metadataRequestHeader, metadataRequest); // <3>
+        return stage.thenApply(metadataResponse -> mutateFetchRequest(metadataResponse, request)) // <4>
+                    .thenCompose(mutatedFetchRequest -> context.forwardRequest(header, mutatedFetchRequest)); // <5>
     }
 }
 ----
-<1> We construct a new request object.  This will be forwarded asynchronously.
-<2> We send the request towards the Cluster, specifying the api version to use and obtaining a CompletionStage for the
-response.
-<3> We use a computation stage to mutate the filtered fetch request using the response from the request sent at <1>.
-<4> We use another computation stage to forward the mutated request.
+<1> We construct a header object for the asynchronous request.  It is important to specify the API version of the request
+that is to be used.  The version chosen must be a version know to the Kafka Client used by Kroxylicious
+and must be an API version supported by the Target Cluster.
+<2> We construct a new request object. When constructing the request object, take care to populate  the fields known to
+the request object at the API version you have chosen.  Refer to the {kafka-protocol}[Kafka Protocol Guide] for more details.
```

Comment:
I think we should be making a stronger statement about alignment with versions. 

```suggestion
<2> We construct a new request object. When constructing the request object, care needs to be taken to ensure the request is populated with the structure which matches the API version you have chosen.  Refer to the {kafka-protocol}[Kafka Protocol Guide] for more details.
```

I'm not sure that actually reads any better however.

Reply:
I'll take your version.